### PR TITLE
Feature/43538 data sync user input

### PR DESCRIPTION
**Pull request by jon-shipley**
Persist user inputs from the SPA into SQL Server

As a part of this PR, the inputs stored in the SPA have been simplified by using the PointerEvents API.

### CODE REVIEW

**Review by GuyHarwood — COMMENTED** [review]
LGSF

File: tslib/src/functions-throttled/sync-results-to-sql/question.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      code: 'Q001'
+    }
+    const mockWarmupQuestion: DBQuestion = {
+      id: 2,
+      factor1: 2,
+      factor2: 2,
+      isWarmup: true,
+      code: 'W002'
+    }
+    jest.spyOn((sut as any), 'sqlGetQuestionData').mockReturnValue(Promise.resolve([mockQuestion, mockWarmupQuestion]))
```
Comment (GuyHarwood) [inline tslib/src/functions-throttled/sync-results-to-sql/question.service.spec.ts:30]: interesting way to mock the injected sql service!

**Review by GuyHarwood — CHANGES_REQUESTED** [review]
LGTM, just the data migrations that need attention.

File: db/migrations/schema-objects/20201030104251.do.add-pen-input-type.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM mtc_results.userInputTypeLookup WHERE code = 'P';
+INSERT INTO mtc_results.userInputTypeLookup (code, name) VALUES ('P', 'Pen');
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply db/migrations/schema-objects/20201030104251.do.add-pen-input-type.sql]: Done!  However, the order was as expected.

**Review by GuyHarwood — APPROVED** [review]
Thanks for the changes

**Pull request merged**